File: Data Structures & Algorithms/sliding-window-maximum/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} k
     * @return {number[]}
     */
    maxSlidingWindow(nums, k) {
        let result = [];

        for(let i=0;i<=nums.length - k;i++){
            let max = nums[i];
            for(let j=i;j<i+k;j++){
                max = Math.max(max, nums[j])
            }
            result.push(max);
        }
        return result;
    }
}

File: Data Structures & Algorithms/duplicate-integer/submission-0.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        unordered_map<int,int> map;
        for(auto x:nums){
            map[x]++;
        }
        for(auto x:nums){
            if(map[x] > 1) return true;
        }
        return false;
    }
};
